docs/BUILDING.md

```diff
@@ -119,6 +119,72 @@ zypper in gcc gcc-c++ git cmake ninja SDL2-devel libpng16-devel libzip-devel lib
 # or using clang
 zypper in clang libstdc++-devel git cmake ninja SDL2-devel libpng16-devel libzip-devel libzip-tools nlohmann_json-devel tinyxml2-devel spdlog-devel
 ```
+#### Nix
+You can use a `flake.nix` file to instantly setup a development environment using [Nix](https://nixos.org/). Write this `flake.nix` file in the root directory:
```

Comment:
would you be willing to add Nix to https://github.com/HarbourMasters/Shipwright/blob/develop/.github/workflows/test-builds-on-distros.yml?

i haven't run/updated it in a while, but it'd be great to be able to make sure we don't break nix next time we either add a dep or do some cmake deps cleanup stuff